txscript/sigcache.go

```diff
@@ -79,35 +84,11 @@ func (s *SigCache) Add(sigHash wire.ShaHash, sig *btcec.Signature, pubKey *btcec
 	// If adding this new entry will put us over the max number of allowed
 	// entries, then evict an entry.
 	if uint(len(s.validSigs)+1) > s.maxEntries {
-		// Generate a cryptographically random hash.
-		randHashBytes := make([]byte, wire.HashSize)
-		_, err := rand.Read(randHashBytes)
-		if err != nil {
-			// Failure to read a random hash results in the proposed
-			// entry not being added to the cache since we are
-			// unable to evict any existing entries.
-			return
-		}
-
-		// Try to find the first entry that is greater than the random
-		// hash. Use the first entry (which is already pseudo random due
-		// to Go's range statement over maps) as a fall back if none of
-		// the hashes in the rejected transactions pool are larger than
-		// the random hash.
-		var foundEntry sigInfo
 		for sigEntry := range s.validSigs {
```

Comment:
Please add a comment that calls out this is relying on the fact that Go iterators maps in a pseudo-random fashion.  People who aren't familiar with that particular nuance will otherwise think this code is just deleting the first entry.